modules/xcode/_preload.lua

```diff
@@ -40,6 +40,10 @@
 		}
 	}
 
+	p.api.addAllowed("debugformat", {
+		"dwarf-with-dsym",
```

Comment:
There's a similar option on Linux that does the same thing, it's called something else though. Given that they're the same thing, but have different names, it would be good to use generic terminology - I believe `dsym` is very much an Apple term, I only got macOS hits when I googled it the other night.

Reply:
The idea here was to not introduce a layer of mapping between premake values and what is generated in the xcode project file (ie: what you enter is what you get).

I (mistakenly) thought that by adding to `debugformat` in the xcode action, those options would not be available outside of the module, whoops!

It looks like the equivalent gcc flag is be `-gsplit-dwarf`. Ill make that the value instead.